base/sort.jl

```diff
@@ -285,14 +285,15 @@ function searchsortedfirst(a::AbstractRange{<:Integer}, x::Real, o::DirectOrderi
         lastindex(a) + 1
     else
         if o isa ForwardOrdering
-            -fld(floor(Integer, -x) + first(a), h) + 1
+            y = isa(x, Unsigned) ? floor(-Signed(x)) : floor(Integer, -x)
```

Comment:
Yes, the method below is used for x::Unsigned. I have restored the original, except for using `Signed(first(a))` instead of `first(a)` (which is needed for passing the tests added in this PR).